test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/BitSetTest.scala

```diff
@@ -0,0 +1,1521 @@
+    bs.set(450)
+    assertArrayEquals(Array[Long](-9223372032559808480L, -9223301668110598271L, 3L, 2L,
+        0L, 0L, 0L, 4L), bs.toLongArray())
+  }
+
+  @Test def valueOf_ByteArray(): Unit = {
+    val bs = new BitSet(500)
+    assertEquals(bs, BitSet.valueOf(Array[Byte]()))
+    bs.set(5)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32)))
+    bs.set(32)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1)))
+    bs.set(63)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1, 0, 0, -128)))
+    bs.set(64)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1, 0, 0, -128, 1)))
+    bs.set(71, 110)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1, 0, 0, -128, -127, -1, -1, -1, -1, 63)))
+    bs.set(127, 130)
+    assertEquals(bs,
+      BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1, 0, 0, -128, -127, -1, -1, -1, -1, 63, 0, -128, 3)))
+    bs.set(193)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1, 0, 0, -128, -127, -1, -1, -1, -1, 63, 0,
+      -128, 3, 0, 0, 0, 0, 0, 0, 0, 2)))
+    bs.set(450)
+    assertEquals(bs, BitSet.valueOf(Array[Byte](32, 0, 0, 0, 1, 0, 0, -128, -127, -1, -1, -1, -1, 63, 0,
+      -128, 3, 0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4)))
+  }
+
+  @Test def valueOf_LongArray(): Unit = {
+    val bs = new BitSet(500)
+    assertEquals(bs, BitSet.valueOf(Array[Long]()))
+    bs.set(5)
+    assertEquals(bs, BitSet.valueOf(Array[Long](32L)))
+    bs.set(32)
+    assertEquals(bs, BitSet.valueOf(Array[Long](4294967328L)))
+    bs.set(63)
+    assertEquals(bs, BitSet.valueOf(Array[Long](-9223372032559808480L)))
+    bs.set(64)
+    assertEquals(bs, BitSet.valueOf(Array[Long](-9223372032559808480L, 1L)))
+    bs.set(71, 110)
+    assertEquals(bs, BitSet.valueOf(Array[Long](-9223372032559808480L, 70368744177537L)))
+    bs.set(127, 130)
+    assertEquals(bs, BitSet.valueOf(Array[Long](-9223372032559808480L, -9223301668110598271L, 3L)))
+    bs.set(193)
+    assertEquals(bs, BitSet.valueOf(Array[Long](-9223372032559808480L, -9223301668110598271L, 3L, 2L)))
+    bs.set(450)
+    assertEquals(bs, BitSet.valueOf(Array[Long](-9223372032559808480L, -9223301668110598271L, 3L, 2L,
+      0L, 0L, 0L, 4L)))
+  }
+
+  @Test def valueOf_ByteBuffer(): Unit = {
+    // Array-wrapped ByteBuffer
+    val emptyBS = new BitSet
+    val emptyBytes = emptyBS.toByteArray()
+
+    assertEquals(emptyBS, BitSet.valueOf(ByteBuffer.wrap(emptyBytes)))
+
+    val eightBS = makeEightBS()
+    val eightBytes = eightBS.toByteArray()
+    assertEquals(eightBS, BitSet.valueOf(ByteBuffer.wrap(eightBytes)))
+
+    val bbWithPosition = ByteBuffer.wrap(192.toByte +: eightBytes)
+    assertEquals(bbWithPosition.get(), 192.toByte) // extra byte
+    assertEquals(bbWithPosition.position(), 1)
+    assertEquals(eightBS, BitSet.valueOf(bbWithPosition))
+    assertEquals(bbWithPosition.position(), 1)
```

Comment:
```suggestion
    assertEquals(192.toByte, bbWithPosition.get()) // extra byte
    assertEquals(1, bbWithPosition.position())
    assertEquals(eightBS, BitSet.valueOf(bbWithPosition))
    assertEquals(1, bbWithPosition.position())
```
The expected value is always on the left.

Same below.

Reply:
@sjrd updated, good find